source/Fleet.cpp

```diff
@@ -244,12 +280,13 @@ const Government *Fleet::GetGovernment() const
 // Choose a fleet to be created during flight, and have it enter the system via jump or planetary departure.
 void Fleet::Enter(const System &system, list<shared_ptr<Ship>> &ships, const Planet *planet) const
 {
-	if(!total || variants.empty())
+	if(!total || (variants.empty() && stockVariants.empty()))
```

Comment:
should this be `if((!total || variants.empty()) && (!stockTotal || stockVariants.empty()))`?
Total and variants non-empty seems needed to select from local variants.
stockTotal and stockVariants non-empty seems needed to select from stock variants.